app/config.go

```diff
@@ -70,6 +71,20 @@ type Config struct {
 	FacebookOauthCredentials    *oauth.Credentials
 }
 
+// Lazy initialize Logger for code using partially initialized Config
+func (c *Config) GetLogger() logrus.FieldLogger {
```

Comment:
This does make Logger a bit of special case (perhaps it is), but hs the least impact on tests that regularly pass semi-initialized config. Since it waits until a logger is needed it also gives the best change ErrorReporter is initialised.

Having it on Config will play nicely with any future desire to expose logging options in `configure` later (like which formatter to use).